### DISCUSSION TITLE
Simple question: can you combine badge-number-increases with local-notifications-showups?

### DISCUSSION THREAD

**Discussion by folsze** [capawesome-team/capacitor-plugins#237] [Q&A]
Can you combine badges with local notifications, such that when receiving a local notification, a red number badge appears/increases on the app icon on the home screen?

I am talking about combining those two plugins:

https://capacitorjs.com/docs/apis/local-notifications

https://github.com/capawesome-team/capacitor-plugins/tree/main/packages/badge

Currently I have it like this, but it doesn't create/increase a badge on the app icon on the home screen
(even though the notification DOES show up):

```
  registerNotificationListener() {
    LocalNotifications.addListener('localNotificationReceived', async () => {
      const badgeCount = await Badge.get();
      await Badge.set({ count: badgeCount.count + 1 });
    });
  }
```

```
  public async scheduleTestNotification() {
    const in20Seconds = new Date(new Date().getTime() + 20 * 1000);
    await LocalNotifications.schedule({
      notifications: [
        {
          id: 20,
          title: "TEST 🦾🤖",
          body: "test",
          schedule: {
            at: in20Seconds,
            repeats: false
          }
        },
      ]
    });
  }
```

```
  async requestPermission() {
    const { display } = await LocalNotifications.requestPermissions();
    const badgePermission = await Badge.requestPermissions();
    return display === 'granted' && badgePermission.display === 'granted';
  }
```

Capacitor v5

**Comment by robingenz** [comment]
I have not tested this. Maybe someone from the community has experience with it.

**Comment by folsze** [comment]
Thanks a lot for your reply, I really appreciate it. Actually now I realized that it shows up with this setup, at some point. Flakey still. Need to investigate a bit more :)
